eva/executor/create_index_executor.py

```diff
@@ -0,0 +1,160 @@
+# coding=utf-8
+# Copyright 2018-2022 EVA
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from copy import deepcopy
+from pathlib import Path
+
+import faiss
+import pandas as pd
+
+from eva.catalog.catalog_manager import CatalogManager
+from eva.catalog.column_type import ColumnType, NdArrayType
+from eva.catalog.index_type import IndexType
+from eva.configuration.constants import EVA_DEFAULT_DIR, INDEX_DIR
+from eva.executor.abstract_executor import AbstractExecutor
+from eva.models.storage.batch import Batch
+from eva.parser.create_statement import ColConstraintInfo, ColumnDefinition
+from eva.planner.create_index_plan import CreateIndexPlan
+from eva.storage.storage_engine import StorageEngine
+from eva.utils.logging_manager import logger
+
+
+class CreateIndexExecutor(AbstractExecutor):
+    def __init__(self, node: CreateIndexPlan):
+        super().__init__(node)
+
+    def validate(self):
+        pass
+
+    def exec(self):
+        catalog_manager = CatalogManager()
+        if catalog_manager.get_index_by_name(self.node.name):
+            msg = f"Index {self.node.name} already exists."
+            logger.error(msg)
+            raise RuntimeError(msg)
+
+        # Get feature tables.
+        table_ref = self.node.table_ref
+        df_metadata = CatalogManager().get_dataset_metadata(
+            table_ref.table.database_name, table_ref.table.table_name
+        )
+
+        # Add features to index.
+        # TODO: batch size is hardcoded for now.
+        index = None
+        input_dim, feat_size = -1, 0
+        for batch in StorageEngine.read(df_metadata, 1):
+            feat = batch.column_as_numpy_array(self.node.col_list[0].name)[0]
```

Comment:
What is going on here?

Reply:
1. Get the feature column name.
2. Read the feature vector for that column.

Reply:
See if you can use this function. 
https://github.com/georgia-tech-db/eva/blob/c35a1990d4716ca0d7cba6d6b857ded6fec6e656/eva/models/storage/batch.py#L241

Reply:
I think the code stays pretty much the same if I use project. I did rework the code piece and add some comments. Pls check if that improves the readability.